core/src/main/scala/kafka/log/Log.scala

```
@@ -1512,6 +1547,18 @@ class Log(@volatile var dir: File,
     }
   }
 
+  /**
+    * Given a message offset, find its corresponding offset metadata in the log.
+    * If the message offset is out of range, throw an OffsetOutOfRangeException
+    */
+  def convertToOffsetMetadataOrThrow(offset: Long, message: String): LogOffsetMetadata = {
```

Seems it would be more natural to let `convertToOffsetMetadata` call this method instead of the other way around? Otherwise we catch and discard the exception only to re-throw a new one.

---

That's actually what I had originally. I'll switch it back